src/System.Net.Requests/tests/HttpWebRequestTest.cs

```diff
@@ -1531,6 +1615,119 @@ public void MediaType_SetThenGet_ValuesMatch(Uri remoteServer)
+                            Assert.False(secondAccept.IsCompleted);
+                            await VerifyResponse(responseContent, secondResponseTask);
+                        }
+                        else
+                        {
+                            await VerifyNewConnection(responseContent, secondAccept, secondResponseTask);
+                        }
+                    }
+                }
+                return RemoteExecutor.SuccessExitCode;
+            }, JsonSerializer.Serialize<HttpWebRequestParameters>(requestParameters), connectionReusedParameter.ToString()).Dispose();
+        }
+
+        [Fact]
+        public void GetResponseAsync_ParametersAreCachableButDifferent_CreateNewClient()
+        {
+            RemoteExecutor.Invoke(async () =>
+             {
+                 using (var listener = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp))
+                 {
+                     listener.Bind(new IPEndPoint(IPAddress.Loopback, 0));
+                     listener.Listen(1);
+                     var ep = (IPEndPoint)listener.LocalEndPoint;
+                     var uri = new Uri($"http://{ep.Address}:{ep.Port}/");
+
+                     var referenceParameters = new HttpWebRequestParameters
+                     {
+                         AllowAutoRedirect = true,
+                         AutomaticDecompression = DecompressionMethods.GZip,
+                         MaximumAutomaticRedirections = 2,
+                         MaximumResponseHeadersLength = 100,
+                         PreAuthenticate = true,
+                         SslProtocols = SecurityProtocolType.Tls12,
+                         Timeout = 100000
+                     };
+                     HttpWebRequest firstRequest = WebRequest.CreateHttp(uri);
+                     referenceParameters.Configure(firstRequest);
+                     firstRequest.Method = HttpMethod.Get.Method;
+
+                     string responseContent = "Test response.";
+
+                     Task<WebResponse> firstResponseTask = firstRequest.GetResponseAsync();
+                     using (Socket server = await listener.AcceptAsync())
+                     using (var serverStream = new NetworkStream(server, ownsSocket: false))
+                     using (var serverReader = new StreamReader(serverStream))
+                     {
+                         await ReplyToClient(responseContent, server, serverReader);
+                         await VerifyResponse(responseContent, firstResponseTask);
+
+                         foreach (var caseRow in CachableWebRequestParameters())
+                         {
+                             var currentParameters = (HttpWebRequestParameters)caseRow[0];
+                             var connectionReused = (bool)caseRow[1];
+                             Task<Socket> secondAccept = listener.AcceptAsync();
+
+                             var currentRequest = WebRequest.CreateHttp(uri);
```

Comment:
Nit: please only use var when the type is evident on the right-hand side (ctors, explicit casts)